tfprotov5/state.go

```diff
@@ -77,3 +77,15 @@ func (s RawState) Unmarshal(typ tftypes.Type) (tftypes.Value, error) {
 	}
 	return tftypes.Value{}, ErrUnknownRawStateType
 }
+
+// UnmarshalWithOpts is identical to Unmarshal but also accepts a tftypes.UnmarshalOpts which contains
+// options that can be used to modify the behaviour when unmarshalling JSON or Flatmap.
+func (s RawState) UnmarshalWithOpts(typ tftypes.Type, opts tftypes.UnmarshalOpts) (tftypes.Value, error) {
```

Comment:
Could you add brief unit tests for these new `UnmarshalWithOpts` functions (similar to the `ValueFromJSONWithOpts` ones)?

We'll also need to ensure all these new pieces are captured in a CHANGELOG entry file. 👍 

Thanks!

Reply:
Added unit tests for `UnmarshalWithOpts`.